client/components/tinymce/plugins/simple-payments/dialog/form.jsx

```diff
@@ -42,6 +51,12 @@ const validate = ( values, props ) => {
 
 	if ( ! values.price ) {
 		errors.price = translate( 'Price can not be empty.' );
+	} else if ( parseFloat( values.price ) === NaN ) {
+		errors.price = translate( 'Price is malformed.' );
```

Comment:
nit: Calypso validation usually says "Price is invalid." instead of "malformed".